microovn/ovn/bootstrap.go

```diff
@@ -58,12 +70,36 @@ func Bootstrap(s *state.State) error {
 		return fmt.Errorf("Failed to start OVS switch: %w", err)
 	}
 
+	// Generate certificate for OVN Central services
+	nbCertPath, nbKeyPath := paths.PkiOvnNbCertFiles()
+	sbCertPath, sbKeyPath := paths.PkiOvnSbCertFiles()
+	northdCertPath, northdKeyPath := paths.PkiOvnNorthdCertFiles()
+	err = GenerateNewServiceCertificate(s, "ovnnb", CertificateTypeServer, nbCertPath, nbKeyPath)
+	if err != nil {
+		return fmt.Errorf("failed to generate TLS certificate for ovnnb service: %s", err)
+	}
+	err = GenerateNewServiceCertificate(s, "ovnsb", CertificateTypeServer, sbCertPath, sbKeyPath)
+	if err != nil {
+		return fmt.Errorf("failed to generate TLS certificate for ovnsb service: %s", err)
+	}
+	err = GenerateNewServiceCertificate(s, "ovn-northd", CertificateTypeClient, northdCertPath, northdKeyPath)
```

Comment:
Any particular reason for using a client-certificate here?

All of the OVN daemons use the same library for DB communication. This library allows specification of any connection to be either active or passive, i.e. while unusual, northd might be told to behave like a server, expecting a connection from the DB server.

Reply:
If the KeyUsage does not contain the right bits for a server, the library will assert when attempting to create a listener, we've hit this issue in the field before. So since it is possible to configure listeners for any program based on the OVS libraries, let's just use server certs everywhere.